module/data_consumer.go

```diff
@@ -0,0 +1,82 @@
+package module
+
+import (
+	"context"
+	"os"
+	"time"
+
+	datapb "go.viam.com/api/app/data/v1"
+
+	"go.viam.com/rdk/app"
+	"go.viam.com/rdk/utils"
+)
+
+// QueryTabularDataOptions provides additional input for QueryTabularDataForResource.
+type QueryTabularDataOptions struct {
+	TimeBack         time.Duration
+	AdditionalStages []map[string]any
+}
+
+// ResourceDataConsumer can be added as an anonymous struct member to a resource to enable historical module data queries.
+type ResourceDataConsumer struct {
+	_dataClient *app.DataClient
```

Comment:
(minor) since `dataClient` isn't a function anymore, I think we can drop the `_` prefix here.